src/main/scala/com/amazon/deequ/checks/Check.scala

```diff
@@ -1092,7 +1124,10 @@ case class Check(
         case nc: ConstraintDecorator => nc.inner
         case c: Constraint => c
       }
-      .collect { case constraint: AnalysisBasedConstraint[_, _, _] => constraint.analyzer }
+      .collect {
+        case constraint: AnalysisBasedConstraint[_, _, _] => constraint.analyzer
+        case constraint: DataSynchronizationConstraint => constraint.analyzer
```

Comment:
Why can't the existing statement match `DataSynchronizationConstraint` ?

Reply:
updated, yes we don't need this.